File: app/src/main/java/com/example/website/consulta/ConsultaActivity.java
package com.example.website.consulta;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import android.widget.Toast;

public class ConsultaActivity extends AppCompatActivity {
    private Spinner spFiltro;
    private FragmentTransaction frManager;
    private Fragment frMotor, frAlternante, frCodBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_consulta);

        frMotor = new MotorFragment();
        frAlternante = new AlternanteFragment();
        frCodBar = new CodbarFragment();

        spFiltro = findViewById(R.id.spFiltro);
        spFiltro.setAdapter(ListaFiltros());
        spFiltro.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
                frManager = getSupportFragmentManager().beginTransaction();
                if (i == 0) {
                    //>  frManager.addToBackStack("");
                    frManager.replace(R.id.frContent, frAlternante);
                    frManager.addToBackStack(null);
                } else if(i == 1) {
                    frManager.replace(R.id.frContent, frCodBar);
                    frManager.addToBackStack(null);
                }
                else{
                    frManager.replace(R.id.frContent, frMotor);
                    frManager.addToBackStack(null);
                }
                frManager.commit();
            }

            @Override
            public void onNothingSelected(AdapterView<?> adapterView) {

            }
        });

        //> getSupportFragmentManager().beginTransaction().add(R.id.frContent, frMotor);
    }

    private ArrayAdapter<CharSequence> ListaFiltros() {
        //> String[] opciones = {"Alternante", "codigo.bar", "Nombre", "Motores"};

        //> ArrayAdapter<String> adapter = new ArrayAdapter<String>(this, R.layout.spinner_item_editor, opciones);
        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this, R.array.consulta_opcion, R.layout.support_simple_spinner_dropdown_item);
        adapter.setDropDownViewResource(R.layout.support_simple_spinner_dropdown_item);
        return adapter;
    }
}

File: app/src/main/java/com/example/website/consulta/View/ConsultaActivity.kt
package com.example.website.consulta.View

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemSelectedListener
import android.widget.ArrayAdapter
import android.widget.Spinner
import com.example.website.consulta.Model.Entidad.Articulo
import com.example.website.consulta.R
import com.example.website.consulta.ViewModel.NVentasViewModel
import kotlinx.android.synthetic.main.activity_factura.*

class ConsultaActivity : AppCompatActivity() {
    private var spFiltro: Spinner? = null
    private lateinit var frManager: FragmentTransaction
    private var frMotor: Fragment? = null
    private var frAlternante: Fragment? = null
    private var frCodBar: Fragment? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_consulta)
        initializeComponents()
        initializeEvents()
    }

    private fun initializeComponents() {
        frMotor = MotorFragment()
        frAlternante = AlternanteFragment()
        frCodBar = CodbarFragment()
        spFiltro = findViewById(R.id.spFiltro)
    }

    private fun initializeEvents() {
        spFiltro?.setAdapter(ListaFiltros())
        spFiltro?.setOnItemSelectedListener(spFiltroOnItemSelectedListener)
    }

    private fun ListaFiltros(): ArrayAdapter<CharSequence> {
        //> String[] opciones = {"Alternante", "codigo.bar", "Nombre", "Motores"};

        //> ArrayAdapter<String> adapter = new ArrayAdapter<String>(this, R.layout.spinner_item_editor, opciones);
        val adapter = ArrayAdapter.createFromResource(
            this,
            R.array.consulta_opcion,
            R.layout.support_simple_spinner_dropdown_item
        )
        adapter.setDropDownViewResource(R.layout.support_simple_spinner_dropdown_item)
        return adapter
    }

    private val spFiltroOnItemSelectedListener = object : AdapterView.OnItemSelectedListener {
        override fun onItemSelected(p0: AdapterView<*>?, p1: View?, i: Int, p3: Long) {
            frManager = supportFragmentManager.beginTransaction()
            frManager.disallowAddToBackStack()
            when (i) {
                0 -> {
                    frManager.replace(R.id.frContent, frAlternante as AlternanteFragment)
                    //> frManager!!.addToBackStack(null)
                }
                1 -> {
                    frManager.replace(R.id.frContent, frCodBar as CodbarFragment)
                    //> frManager!!.addToBackStack(null)
                }
                else -> {
                    frManager.replace(R.id.frContent, frMotor as MotorFragment)
                    //> frManager!!.addToBackStack(null)
                }
            }
            frManager.commit()
        }

        override fun onNothingSelected(p0: AdapterView<*>?) {

        }
    }
}